### PR TITLE
[record-minmax] Revise record-minmax

### PR DESCRIPTION
**Pull request by jinevening**
This introduces MinMaxComputer to easily update calibration logic.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: #11269 
Draft PR: https://github.com/Samsung/ONE/pull/11409

### CODE REVIEW

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 1]
```diff
@@ -144,6 +143,8 @@ void verifyTypeShape(const luci::CircleInput *input_node, const DataType &dtype,
   }
 }
 
+// TODO Remove unused code
+#if 0
 void update_quantparam(record_minmax::MinMaxObserver *observer, const std::string &mode,
```
Comment (jinevening) [inline compiler/record-minmax/src/RecordMinMax.cpp:148]: ## Motivation behind this PR

Previously, this local function computes min/max values for all algorithms. This makes it is difficult to update an algorithm, e.g., adding more hyperparameters (#11269).

This PR introduces MinMaxComputer to tackle the issue. After this PR, we may only update MinMaxComputer to change min/max computation logic.

File: compiler/record-minmax/include/MinMaxComputer.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+public:
+  PercentileComputer(float min_percentile, float max_percentile)
+    : _min_percentile(min_percentile), _max_percentile(max_percentile)
+  {
+  }
+
+  virtual void update_qparam(MinMaxObserver *observer);
+
+private:
+  float _min_percentile;
```
Comment (seanshpark) [inline compiler/record-minmax/include/MinMaxComputer.h:51]: ` = 0.0` ?

**Comment by seanshpark** [comment]
any unit test planned?

**Comment by jinevening** [comment]
> any unit test planned?

The core logic is tested in `RecordFunction.test.cpp`. But I'll add unit tests to increase code coverage. The signature of `update_qparam` will change, so I'll add tests after that.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**